app/subscribers/solidus_stripe/webhook/payment_intent_subscriber.rb

```diff
@@ -27,6 +27,32 @@ def complete_payment(event)
           )
         end
       end
+
+      # Voids a payment.
+      #
+      # Marks a Solidus payment associated to a Stripe payment intent as
+      # failed, adding a log entry about the event.
+      #
+      # @param event [SolidusStripe::Webhook::Event]
+      def fail_payment(event)
+        payment = extract_payment_from_event(event)
+        return if payment.failed?
```

Comment:
```suggestion
        return if !payment.can_failure?
```

I'd say that checking more generally if we can transition to failed is more generic and safe. You could be on a state (like checkout) where payment is not failed but doing the transition will raise an error. Same goes for the complete! event of the same file.

Reply:
Thanks for your suggestion, @loicginoux. If Solidus payment is in the `:checkout` state, but we're already getting a `payment_intent.payment_failed` webhook, that's probably an abnormal situation. I think it's better to be stricter so we can't detect any failure in our integration.